Reactivate CI on custom_report

fcayre opened the pull request:
...  thanks to new account_usability fork, with setuptools config included.

fcayre committed:
Fix tests and dependencies

fcayre committed:
Reactivate CI on custom_report

...  thanks to new account_usability fork, with setuptools config included.

fcayre commented:
@legalsylvain je suis preneur de tes lumières sur ce coup-là, pour la CI ne trouve pas https://github.com/OCA/server-tools/tree/12.0/base_view_inheritance_extension ?

legalsylvain commented:
Hello ! 

alors ça bosse le dimanche ;-) ?

> @legalsylvain je suis preneur de tes lumières sur ce coup-là, pour la CI ne trouve pas https://github.com/OCA/server-tools/tree/12.0/base_view_inheritance_extension ?

Ce n'est pas un module OCA. La CI ne peut pas la trouver automatiquement via pypi. C'est un module akretion aussi : 
- https://github.com/akretion/odoo-usability/tree/12.0/base_company_extension

Du coup, si tu rajoutes cette ligne, ça "devrait" le faire : 

``git+https://github.com/fcayre/odoo-usability@12.0-fix-account_usability-and-make-pip-installable#subdirectory=setup/base_company_extension``

Bon dimanche.

fcayre commented:
Non je bosse jamais le dimanche moi, et toi :-p ?

Note j'ai pas les yeux en face des trous parce que je cherche base_view_inheritance_extension au lieu de base_company_extension... Merci !

fcayre committed:
Fix copyright and such

legalsylvain commented on requirements.txt:
```diff
@@ -3,3 +3,5 @@ iso8601
 phonenumbers
 python-magic
 requests
+py3o.template
```
Il ne faut pas les ajouter ici, mais dans le fichier "__manifest__.py".
Le fichier requirements est autogenere a partir de tous les manifestes.

legalsylvain replied:
Generation faite par ton pre-commit by the way.

legalsylvain replied:
Nota : si par hasard, le import n'a pas la même valeur que le pip install tu peux mettre la valeur de l'import dans le fichier ``__manifest__.py`` et la valeur du pip install dans le fichier ``setup/MODULE_NAME/setup.py``

Exemple pour magic / python-magic : 
- import = ``magic`` : https://github.com/commown/commown-odoo-addons/blob/12.0/commown/__manifest__.py#L55
- pip install = ``python-magic`` : https://github.com/commown/commown-odoo-addons/blob/12.0/setup/commown/setup.py#L6

(me suis bien galéré avant de comprendre ce hack)

fcayre replied:
Merci !

legalsylvain commented on custom_report/__manifest__.py:
```diff
@@ -10,17 +10,21 @@
+    'author': "Commown SCIC",
     'license': "AGPL-3",
-    'website': "https://commown.fr",
+    'website': "https://commown.coop",
```
et hop ! j'ai activé un truc dans le pre-commit qui corrige automatiquement la valeur de website.

https://github.com/commown/commown-odoo-addons/pull/23

fcayre committed:
Fix report and tests wrt 12.0 API changes

In the report:
- analytic_account_id has been replaced by contract_line_id.contract_id on invoice lines
- dict.values() must be replaced by list(dict.values()) (py3 related change)

In the tests:
- selected partner's xml id was no more a b2c partner
- contract and invoice are now related by their lines.

fcayre commented:
LOL @legalsylvain je comprends encore mieux ton avertissement de vendredi concernant l10n_generic_coa et le hook post init de account : ici les tests lancés dans une base de données fabriquée pour un seul module (commown_cooperative_campaign) passent parfaitement mais plantent lancés dans la CI... je pense parce que la taxe par défaut n'est pas la même dans les deux cas.

En effet custom_report dépend de l10n_fr (pour utiliser le siret dans le rapport), et je pense que la taxe US à 15% n'est plus la taxe par défaut. Bon ben je verra ça demain, là je suis crevé :-(

legalsylvain commented:
;-) 
Oui, grâce à ``l10n_generic_coa``, la CI peut contenir des tests de Schrodinger.

C'est assez agaçant ! 

Bon courage.

fcayre committed:
Use untaxed amounts in tests to avoid l10n_generic_coa post init hook oddities

legalsylvain commented:
et c'est vert ! bravo.
